tests/test_main_check_connection.py

```
@@ -0,0 +1,612 @@
+    created_tasks: list[_TaskSpy] = []
+    real_create_task = asyncio.create_task
+
+    def _capture_create_task(coro: Any, *args: Any, **kwargs: Any) -> Any:
+        task = real_create_task(coro, *args, **kwargs)
+        spy = _TaskSpy(task)
+        created_tasks.append(spy)
+        return spy
+
+    real_wait_for = asyncio.wait_for
+
+    def _awaitable_name(awaitable: Any) -> str:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `awaitable`

(ANN401)